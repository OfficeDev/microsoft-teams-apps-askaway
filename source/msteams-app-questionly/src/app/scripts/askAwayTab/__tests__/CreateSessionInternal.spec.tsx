/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, Button, Form, Input, TextArea } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import CreateSessionInternal from '../popups/CreateSessionInternal';

configure({ adapter: new Adapter() });
jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
        };
    },
}));

describe('CreateSessionInternal Component', () => {
    const onSubmitCreateSession = jest.fn();

    afterAll(() => {
        jest.resetAllMocks();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<CreateSessionInternal onSubmitCreateSession={onSubmitCreateSession} />);

        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<CreateSessionInternal onSubmitCreateSession={onSubmitCreateSession} />);

        expect(component.find(Flex)).toHaveLength(2);
        expect(component.find(Form)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(2);
        expect(component.find(Input)).toHaveLength(1);
        expect(component.find(TextArea)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
    });
});
