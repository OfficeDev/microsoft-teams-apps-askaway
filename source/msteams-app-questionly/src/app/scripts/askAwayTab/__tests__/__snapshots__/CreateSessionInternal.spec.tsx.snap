// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateSessionInternal Component should match the snapshot 1`] = `
<div
  style={
    Object {
      "padding": "1rem 2rem",
    }
  }
>
  <Flex
    column={true}
  >
    <Form
      as="form"
      onSubmit={[Function]}
    >
      <div
        className="form-grid"
      >
        <Text
          as="span"
          content="popups.sessionTitle"
          size="small"
        />
        <Input
          accessibility={[Function]}
          as="div"
          errorIndicator={<ExclamationCircleIcon />}
          fluid={true}
          iconPosition="end"
          label=""
          maxLength={250}
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="popups.sessionTitlePlaceholder"
          successIndicator={<PresenceAvailableIcon />}
          type="text"
          wrapper={Object {}}
        />
      </div>
      <div
        className="form-grid"
      >
        <Text
          as="span"
          content="popups.sessionDescription"
          size="small"
        />
        <TextArea
          accessibility={[Function]}
          as="textarea"
          fluid={true}
          maxLength={250}
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="popups.sessionDescriptionPlaceholder"
          styles={
            Object {
              "marginTop": "0.25rem",
            }
          }
        />
      </div>
      <div
        className="form-grid"
      >
        <Flex
          hAlign="end"
          vAlign="end"
        >
          <Button
            accessibility={[Function]}
            as="button"
            className="btn-create-session"
            primary={true}
            size="medium"
            type="submit"
          >
            <ButtonContent
              as="span"
            >
              popups.createQnaSessionButton
            </ButtonContent>
          </Button>
        </Flex>
      </div>
    </Form>
  </Flex>
</div>
`;
